peps/pep-0705.rst

```diff
@@ -310,235 +217,138 @@ This is consistent with the behavior introduced in :pep:`655`.
 
-Subclasses can also require keys that are read-only but not required in the superclass::
+Subclasses can require items that are read-only but not required in the superclass::
 
-    class OptionalName(TypedDict, readonly=True):
-        name: NotRequired[str]
+    class OptionalName(TypedDict):
+        name: ReadOnly[NotRequired[str]]
 
-    class Person(OptionalName, TypedDict):
-        name: Required[str]
+    class RequiredName(OptionalName):
+        name: ReadOnly[Required[str]]
 
-    person: Person = {}  # Type check error: "name" required
+    d: RequiredName = {}  # Type check error: "name" required
 
-Note that these are just consequences of structural typing, but they are highlighted here as the behavior now differs from the rules specified in :pep:`589`.
+Subclasses can combine these rules::
 
-Finally, subclasses can have ``other_keys=Never`` even if the superclass does not::
+    class OptionalIdent(TypedDict):
+        ident: ReadOnly[NotRequired[str | int]]
 
-    class Person(OptionalName, other_keys=Never):
-        name: Required[str]
+    class User(OptionalIdent):
+        ident: str  # Required, mutable, and not an int
+
+Note that these are just consequences of structural typing, but they are highlighted here as the behavior now differs from the rules specified in :pep:`589`.
 
 Type consistency
 ----------------
 
-*This section updates the type consistency rules introduced in* :pep:`589` *to cover the new features in this PEP. In particular, any pair of types that do not use the new features will be consistent under these new rules if (and only if) they were already consistent.*
-
-A TypedDict type with ``other_keys=Never`` is consistent with ``Mapping[str, V]``, where ``V`` is the union of all its value types. For instance, the following type is consistent with ``Mapping[str, int | str]``::
-
-    class Person(TypedDict, other_keys=Never):
-        name: str
-        age: int
+*This section updates the type consistency rules introduced in* :pep:`589` *to cover the new feature in this PEP. In particular, any pair of types that do not use the new feature will be consistent under these new rules if (and only if) they were already consistent.*
 
 A TypedDict type ``A`` is consistent with TypedDict ``B`` if ``A`` is structurally compatible with ``B``. This is true if and only if all of the following are satisfied:
 
-* For each key in ``B``, ``A`` has the corresponding key and the corresponding value type in ``A`` is consistent with the value type in ``B``, unless the key in ``B`` is of type ``ReadOnly[NotRequired[Any]]``, in which case it may be missing in ``A`` provided ``A`` allows other keys.
-* For each non-read-only key in ``B``, the corresponding value type in ``B`` is also consistent with the corresponding value type in ``A``.
+* For each item in ``B``, ``A`` has the corresponding key, unless the item in ``B`` is read-only, not required, and of top value type (e.g. ``ReadOnly[NotRequired[Any]]``).
```

Comment:
@erictraut Should I have changed this `Any` to `object`?

Reply:
Yes, I think this should be `ReadOnly[NotRequired[object]]`. `object` is the top type in the Python type system.